### ISSUE TITLE
Extension stopped working after lockin the screen

### ISSUE THREAD

**Issue by Razer0123** [mlutfy/hidetopbar#360]
I got this error

Main.layoutManager.primaryMonitor is null

Any way to fix this?

**Comment by tuxor1337** [comment]
Please provide information about your setup:
1. Operating system.
2. Version of GNOME.
3. Version of Hide Top Bar.
4. Your Hide Top Bar settings.
5. Extensions running in parallel.
6. Number of monitors.

**Comment by Razer0123** [comment]
1.     Fedora 35 
2.     41.5
3.     107
4.     Default + show panel only when mouse is close to edge of screen
5.     Pop shell, Tray icons Reloaded, Weatherin the clock, just perfection, freon, emoji selector, blur my shell, brightness color using ddcutil
6.     1

**Comment by Razer0123** [comment]
Any news?

**Comment by tuxor1337** [comment]
I can't reproduce the problem on my setup (GNOME 42 on Fedora 36). It might well be due to other extensions running in parallel. Can you reproduce it with all other extensions deactivated? Apart from that, there is not much I can do as long as I can't reproduce the problem.

**Comment by Razer0123** [comment]
> I can't reproduce the problem on my setup (GNOME 42 on Fedora 36). It might well be due to other extensions running in parallel. Can you reproduce it with all other extensions deactivated? Apart from that, there is not much I can do as long as I can't reproduce the problem.

I will try,how can i get a proper log to know more about the error?

**Comment by tuxor1337** [comment]
Open a terminal and run `journalctl /usr/bin/gnome-shell -f -o cat`. With the terminal still open, reproduce the problem. Maybe there will be something helpfull in the log output.

**Comment by Razer0123** [comment]
Here is the output, i think i had a similar problem with another extension and it is fixed now, if you find it useful here is the discussion

https://github.com/aunetx/blur-my-shell/issues/137#event-5859858523=

`[alessandro@fedora ~]$ journalctl /usr/bin/gnome-shell -f -o cat
clutter_actor_set_child_below_sibling: assertion 'sibling->priv->parent == self' failed
clutter_actor_set_child_below_sibling: assertion 'sibling->priv->parent == self' failed
clutter_actor_set_child_below_sibling: assertion 'sibling->priv->parent == self' failed
Can't update stage views actor <unnamed>[<MetaWindowGroup>:0x55ddcba46360] is on because it needs an allocation.
Can't update stage views actor <unnamed>[<MetaWindowActorX11>:0x55ddcdc3b710] is on because it needs an allocation.
Can't update stage views actor <unnamed>[<MetaSurfaceActorX11>:0x55ddcdc3fbe0] is on because it needs an allocation.
meta_window_raise: assertion '!window->override_redirect' failed
meta_window_unminimize: assertion '!window->override_redirect' failed
meta_window_activate: assertion '!window->override_redirect' failed
clutter_actor_set_child_below_sibling: assertion 'sibling->priv->parent == self' failed

Can't update stage views actor <unnamed>[<MetaWindowGroup>:0x55ddcba46360] is on because it needs an allocation.
Can't update stage views actor <unnamed>[<MetaWindowActorX11>:0x55ddcdc3b710] is on because it needs an allocation.
Can't update stage views actor <unnamed>[<MetaSurfaceActorX11>:0x55ddcdc3fbe0] is on because it needs an allocation.
Can't update stage views actor <unnamed>[<MetaWindowActorX11>:0x55ddcefb3720] is on because it needs an allocation.
Can't update stage views actor <unnamed>[<MetaSurfaceActorX11>:0x55ddcdc3f860] is on because it needs an allocation.
Can't update stage views actor <unnamed>[<MetaWindowGroup>:0x55ddcba46360] is on because it needs an allocation.
Can't update stage views actor <unnamed>[<MetaWindowActorX11>:0x55ddcefb2f40] is on because it needs an allocation.
Can't update stage views actor <unnamed>[<MetaSurfaceActorX11>:0x55ddcdc3f4e0] is on because it needs an allocation.
JS ERROR: TypeError: menuInstance is null
_updateMenuVisibility@/home/alessandro/.local/share/gnome-shell/extensions/sound-output-device-chooser@kgshank.net/extension.js:210:51
enable/<@/home/alessandro/.local/share/gnome-shell/extensions/sound-output-device-chooser@kgshank.net/extension.js:180:45
_updateVisibility@resource:///org/gnome/shell/ui/status/volume.js:111:9
_maybeShowInput@resource:///org/gnome/shell/ui/status/volume.js:311:14
destroy@/home/alessandro/.local/share/gnome-shell/extensions/sound-output-device-chooser@kgshank.net/extension.js:125:21
disable@/home/alessandro/.local/share/gnome-shell/extensions/sound-output-device-chooser@kgshank.net/extension.js:278:38
_callExtensionDisable@resource:///org/gnome/shell/ui/extensionSystem.js:108:32
_disableAllExtensions/<@resource:///org/gnome/shell/ui/extensionSystem.js:619:22
_disableAllExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:618:52
_sessionUpdated@resource:///org/gnome/shell/ui/extensionSystem.js:636:18
_emit@resource:///org/gnome/gjs/modules/core/_signals.js:114:47
_sync@resource:///org/gnome/shell/ui/sessionMode.js:198:14
pushMode@resource:///org/gnome/shell/ui/sessionMode.js:159:14
activate@resource:///org/gnome/shell/ui/screenShield.js:604:34
lock@resource:///org/gnome/shell/ui/screenShield.js:648:14
LockAsync@resource:///org/gnome/shell/ui/shellDBus.js:504:28
_handleMethodCall@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:354:35
_wrapJSObject/<@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:387:34

display-brightness-ddcutil extension:
Remove keyboard shortcuts
g_signal_handler_disconnect: assertion 'handler_id > 0' failed
g_signal_handler_disconnect: assertion 'handler_id > 0' failed
Source ID 2489 was not found when attempting to remove it
Source ID 2491 was not found when attempting to remove it
Source ID 2493 was not found when attempting to remove it
Source ID 2494 was not found when attempting to remove it
Source ID 5570 was not found when attempting to remove it
Source ID 5571 was not found when attempting to remove it
Source ID 9477 was not found when attempting to remove it
Source ID 11662 was not found when attempting to remove it
Source ID 14038 was not found when attempting to remove it
cr_parser_new_from_buf: assertion 'a_buf && a_len' failed
cr_declaration_parse_list_from_buf: assertion 'parser' failed
loading user theme: /usr/share/themes/Arc-Dark/gnome-shell/gnome-shell.css
Usage of object.actor is deprecated for MarketIndicatorView
get@resource:///org/gnome/shell/ui/environment.js:366:29
_initLayout@/home/alessandro/.local/share/gnome-shell/extensions/bitcoin-markets@ottoallmendinger.github.com/extension.js:2191:13
_init@/home/alessandro/.local/share/gnome-shell/extensions/bitcoin-markets@ottoallmendinger.github.com/extension.js:2164:18
_updateIndicators/indicators<@/home/alessandro/.local/share/gnome-shell/extensions/bitcoin-markets@ottoallmendinger.github.com/extension.js:2343:28
_updateIndicators@/home/alessandro/.local/share/gnome-shell/extensions/bitcoin-markets@ottoallmendinger.github.com/extension.js:2342:47
tryUpdateIndicators@/home/alessandro/.local/share/gnome-shell/extensions/bitcoin-markets@ottoallmendinger.github.com/extension.js:2278:26
IndicatorCollection@/home/alessandro/.local/share/gnome-shell/extensions/bitcoin-markets@ottoallmendinger.github.com/extension.js:2285:13
enable@/home/alessandro/.local/share/gnome-shell/extensions/bitcoin-markets@ottoallmendinger.github.com/extension.js:2368:36
_callExtensionEnable@resource:///org/gnome/shell/ui/extensionSystem.js:167:32
_enableAllExtensions/<@resource:///org/gnome/shell/ui/extensionSystem.js:607:22
_enableAllExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:606:37
_sessionUpdated@resource:///org/gnome/shell/ui/extensionSystem.js:634:18
_emit@resource:///org/gnome/gjs/modules/core/_signals.js:114:47
_sync@resource:///org/gnome/shell/ui/sessionMode.js:198:14
popMode@resource:///org/gnome/shell/ui/sessionMode.js:166:14
_continueDeactivate@resource:///org/gnome/shell/ui/screenShield.js:532:30
deactivate/<@resource:///org/gnome/shell/ui/screenShield.js:523:44
finish@resource:///org/gnome/shell/gdm/authPrompt.js:604:13
finish@resource:///org/gnome/shell/ui/unlockDialog.js:886:26
deactivate@resource:///org/gnome/shell/ui/screenShield.js:523:26
injections_add/<@/usr/share/gnome-shell/extensions/pop-shell@system76.com/extension.js:485:30
ScreenShield/</<@resource:///org/gnome/shell/ui/screenShield.js:109:57
_emit@resource:///org/gnome/gjs/modules/core/_signals.js:114:47
_convertToNativeSignal@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:152:19

Usage of object.actor is deprecated for MarketIndicatorView
get@resource:///org/gnome/shell/ui/environment.js:366:29
_initLayout@/home/alessandro/.local/share/gnome-shell/extensions/bitcoin-markets@ottoallmendinger.github.com/extension.js:2191:13
_init@/home/alessandro/.local/share/gnome-shell/extensions/bitcoin-markets@ottoallmendinger.github.com/extension.js:2164:18
_updateIndicators/indicators<@/home/alessandro/.local/share/gnome-shell/extensions/bitcoin-markets@ottoallmendinger.github.com/extension.js:2343:28
_updateIndicators@/home/alessandro/.local/share/gnome-shell/extensions/bitcoin-markets@ottoallmendinger.github.com/extension.js:2342:47
tryUpdateIndicators@/home/alessandro/.local/share/gnome-shell/extensions/bitcoin-markets@ottoallmendinger.github.com/extension.js:2278:26
IndicatorCollection@/home/alessandro/.local/share/gnome-shell/extensions/bitcoin-markets@ottoallmendinger.github.com/extension.js:2285:13
enable@/home/alessandro/.local/share/gnome-shell/extensions/bitcoin-markets@ottoallmendinger.github.com/extension.js:2368:36
_callExtensionEnable@resource:///org/gnome/shell/ui/extensionSystem.js:167:32
_enableAllExtensions/<@resource:///org/gnome/shell/ui/extensionSystem.js:607:22
_enableAllExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:606:37
_sessionUpdated@resource:///org/gnome/shell/ui/extensionSystem.js:634:18
_emit@resource:///org/gnome/gjs/modules/core/_signals.js:114:47
_sync@resource:///org/gnome/shell/ui/sessionMode.js:198:14
popMode@resource:///org/gnome/shell/ui/sessionMode.js:166:14
_continueDeactivate@resource:///org/gnome/shell/ui/screenShield.js:532:30
deactivate/<@resource:///org/gnome/shell/ui/screenShield.js:523:44
finish@resource:///org/gnome/shell/gdm/authPrompt.js:604:13
finish@resource:///org/gnome/shell/ui/unlockDialog.js:886:26
deactivate@resource:///org/gnome/shell/ui/screenShield.js:523:26
injections_add/<@/usr/share/gnome-shell/extensions/pop-shell@system76.com/extension.js:485:30
ScreenShield/</<@resource:///org/gnome/shell/ui/screenShield.js:109:57
_emit@resource:///org/gnome/gjs/modules/core/_signals.js:114:47
_convertToNativeSignal@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:152:19

Usage of object.actor is deprecated for MarketIndicatorView
get@resource:///org/gnome/shell/ui/environment.js:366:29
_initLayout@/home/alessandro/.local/share/gnome-shell/extensions/bitcoin-markets@ottoallmendinger.github.com/extension.js:2191:13
_init@/home/alessandro/.local/share/gnome-shell/extensions/bitcoin-markets@ottoallmendinger.github.com/extension.js:2164:18
_updateIndicators/indicators<@/home/alessandro/.local/share/gnome-shell/extensions/bitcoin-markets@ottoallmendinger.github.com/extension.js:2343:28
_updateIndicators@/home/alessandro/.local/share/gnome-shell/extensions/bitcoin-markets@ottoallmendinger.github.com/extension.js:2342:47
tryUpdateIndicators@/home/alessandro/.local/share/gnome-shell/extensions/bitcoin-markets@ottoallmendinger.github.com/extension.js:2278:26
IndicatorCollection@/home/alessandro/.local/share/gnome-shell/extensions/bitcoin-markets@ottoallmendinger.github.com/extension.js:2285:13
enable@/home/alessandro/.local/share/gnome-shell/extensions/bitcoin-markets@ottoallmendinger.github.com/extension.js:2368:36
_callExtensionEnable@resource:///org/gnome/shell/ui/extensionSystem.js:167:32
_enableAllExtensions/<@resource:///org/gnome/shell/ui/extensionSystem.js:607:22
_enableAllExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:606:37
_sessionUpdated@resource:///org/gnome/shell/ui/extensionSystem.js:634:18
_emit@resource:///org/gnome/gjs/modules/core/_signals.js:114:47
_sync@resource:///org/gnome/shell/ui/sessionMode.js:198:14
popMode@resource:///org/gnome/shell/ui/sessionMode.js:166:14
_continueDeactivate@resource:///org/gnome/shell/ui/screenShield.js:532:30
deactivate/<@resource:///org/gnome/shell/ui/screenShield.js:523:44
finish@resource:///org/gnome/shell/gdm/authPrompt.js:604:13
finish@resource:///org/gnome/shell/ui/unlockDialog.js:886:26
deactivate@resource:///org/gnome/shell/ui/screenShield.js:523:26
injections_add/<@/usr/share/gnome-shell/extensions/pop-shell@system76.com/extension.js:485:30
ScreenShield/</<@resource:///org/gnome/shell/ui/screenShield.js:109:57
_emit@resource:///org/gnome/gjs/modules/core/_signals.js:114:47
_convertToNativeSignal@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:152:19

display-brightness-ddcutil extension:
Adding to panel
display-brightness-ddcutil extension:
Add keyboard shortcuts
Device did not have an appropriate card
Device did not have an appropriate card
Usage of object.actor is deprecated for PanelMenuButton
get@resource:///org/gnome/shell/ui/environment.js:366:29
EmojisMenu@/home/alessandro/.local/share/gnome-shell/extensions/emoji-selector@maestroschan.fr/extension.js:232:4
enable@/home/alessandro/.local/share/gnome-shell/extensions/emoji-selector@maestroschan.fr/extension.js:452:18
_callExtensionEnable@resource:///org/gnome/shell/ui/extensionSystem.js:167:32
_enableAllExtensions/<@resource:///org/gnome/shell/ui/extensionSystem.js:607:22
_enableAllExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:606:37
_sessionUpdated@resource:///org/gnome/shell/ui/extensionSystem.js:634:18
_emit@resource:///org/gnome/gjs/modules/core/_signals.js:114:47
_sync@resource:///org/gnome/shell/ui/sessionMode.js:198:14
popMode@resource:///org/gnome/shell/ui/sessionMode.js:166:14
_continueDeactivate@resource:///org/gnome/shell/ui/screenShield.js:532:30
deactivate/<@resource:///org/gnome/shell/ui/screenShield.js:523:44
finish@resource:///org/gnome/shell/gdm/authPrompt.js:604:13
finish@resource:///org/gnome/shell/ui/unlockDialog.js:886:26
deactivate@resource:///org/gnome/shell/ui/screenShield.js:523:26
injections_add/<@/usr/share/gnome-shell/extensions/pop-shell@system76.com/extension.js:485:30
ScreenShield/</<@resource:///org/gnome/shell/ui/screenShield.js:109:57
_emit@resource:///org/gnome/gjs/modules/core/_signals.js:114:47
_convertToNativeSignal@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:152:19

Usage of object.actor is deprecated for PanelMenuButton
get@resource:///org/gnome/shell/ui/environment.js:366:29
EmojisMenu@/home/alessandro/.local/share/gnome-shell/extensions/emoji-selector@maestroschan.fr/extension.js:233:4
enable@/home/alessandro/.local/share/gnome-shell/extensions/emoji-selector@maestroschan.fr/extension.js:452:18
_callExtensionEnable@resource:///org/gnome/shell/ui/extensionSystem.js:167:32
_enableAllExtensions/<@resource:///org/gnome/shell/ui/extensionSystem.js:607:22
_enableAllExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:606:37
_sessionUpdated@resource:///org/gnome/shell/ui/extensionSystem.js:634:18
_emit@resource:///org/gnome/gjs/modules/core/_signals.js:114:47
_sync@resource:///org/gnome/shell/ui/sessionMode.js:198:14
popMode@resource:///org/gnome/shell/ui/sessionMode.js:166:14
_continueDeactivate@resource:///org/gnome/shell/ui/screenShield.js:532:30
deactivate/<@resource:///org/gnome/shell/ui/screenShield.js:523:44
finish@resource:///org/gnome/shell/gdm/authPrompt.js:604:13
finish@resource:///org/gnome/shell/ui/unlockDialog.js:886:26
deactivate@resource:///org/gnome/shell/ui/screenShield.js:523:26
injections_add/<@/usr/share/gnome-shell/extensions/pop-shell@system76.com/extension.js:485:30
ScreenShield/</<@resource:///org/gnome/shell/ui/screenShield.js:109:57
_emit@resource:///org/gnome/gjs/modules/core/_signals.js:114:47
_convertToNativeSignal@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:152:19

Timelines with detached actors are not supported
Timelines with detached actors are not supported
Timelines with detached actors are not supported
Timelines with detached actors are not supported
Timelines with detached actors are not supported
Timelines with detached actors are not supported
Timelines with detached actors are not supported
JS ERROR: Extension hidetopbar@mathieu.bidon.ca: TypeError: Main.layoutManager.primaryMonitor is null
_updateSearchEntryMargin@/home/alessandro/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/panelVisibilityManager.js:323:23
_updateSettingsShowInOverview@/home/alessandro/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/panelVisibilityManager.js:318:14
HideTopBar_PanelVisibilityManager@/home/alessandro/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/panelVisibilityManager.js:71:14
enable@/home/alessandro/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/extension.js:38:18
_callExtensionEnable@resource:///org/gnome/shell/ui/extensionSystem.js:167:32
_enableAllExtensions/<@resource:///org/gnome/shell/ui/extensionSystem.js:607:22
_enableAllExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:606:37
_sessionUpdated@resource:///org/gnome/shell/ui/extensionSystem.js:634:18
_emit@resource:///org/gnome/gjs/modules/core/_signals.js:114:47
_sync@resource:///org/gnome/shell/ui/sessionMode.js:198:14
popMode@resource:///org/gnome/shell/ui/sessionMode.js:166:14
_continueDeactivate@resource:///org/gnome/shell/ui/screenShield.js:532:30
deactivate/<@resource:///org/gnome/shell/ui/screenShield.js:523:44
finish@resource:///org/gnome/shell/gdm/authPrompt.js:604:13
finish@resource:///org/gnome/shell/ui/unlockDialog.js:886:26
deactivate@resource:///org/gnome/shell/ui/screenShield.js:523:26
injections_add/<@/usr/share/gnome-shell/extensions/pop-shell@system76.com/extension.js:485:30
ScreenShield/</<@resource:///org/gnome/shell/ui/screenShield.js:109:57
_emit@resource:///org/gnome/gjs/modules/core/_signals.js:114:47
_convertToNativeSignal@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:152:19

display-brightness-ddcutil extension:
ddcutil brief info:
No displays found

meta_monitor_manager_get_logical_monitor_from_number: assertion '(unsigned int) number < g_list_length (manager->logical_monitors)' failed
meta_workspace_get_work_area_for_monitor: assertion 'logical_monitor != NULL' failed
meta_monitor_manager_get_logical_monitor_from_number: assertion '(unsigned int) number < g_list_length (manager->logical_monitors)' failed
meta_workspace_get_work_area_for_monitor: assertion 'logical_monitor != NULL' failed
meta_monitor_manager_get_logical_monitor_from_number: assertion '(unsigned int) number < g_list_length (manager->logical_monitors)' failed
meta_workspace_get_work_area_for_monitor: assertion 'logical_monitor != NULL' failed
meta_monitor_manager_get_logical_monitor_from_number: assertion '(unsigned int) number < g_list_length (manager->logical_monitors)' failed
meta_workspace_get_work_area_for_monitor: assertion 'logical_monitor != NULL' failed
meta_monitor_manager_get_logical_monitor_from_number: assertion '(unsigned int) number < g_list_length (manager->logical_monitors)' failed
meta_workspace_get_work_area_for_monitor: assertion 'logical_monitor != NULL' failed
meta_monitor_manager_get_logical_monitor_from_number: assertion '(unsigned int) number < g_list_length (manager->logical_monitors)' failed
meta_workspace_get_work_area_for_monitor: assertion 'logical_monitor != NULL' failed
cogl_texture_2d_new_with_size: assertion 'width >= 1' failed
cogl_texture_2d_new_with_size: assertion 'width >= 1' failed
cogl_texture_2d_new_with_size: assertion 'width >= 1' failed
cogl_texture_2d_new_with_size: assertion 'width >= 1' failed
Window manager warning: Overwriting existing binding of keysym 65 with keysym 65 (keycode 1a).
cogl_texture_2d_new_with_size: assertion 'width >= 1' failed
meta_monitor_manager_get_logical_monitor_from_number: assertion '(unsigned int) number < g_list_length (manager->logical_monitors)' failed
meta_workspace_get_work_area_for_monitor: assertion 'logical_monitor != NULL' failed
meta_monitor_manager_get_logical_monitor_from_number: assertion '(unsigned int) number < g_list_length (manager->logical_monitors)' failed
meta_workspace_get_work_area_for_monitor: assertion 'logical_monitor != NULL' failed
meta_monitor_manager_get_logical_monitor_from_number: assertion '(unsigned int) number < g_list_length (manager->logical_monitors)' failed
meta_workspace_get_work_area_for_monitor: assertion 'logical_monitor != NULL' failed
cogl_texture_2d_new_with_size: assertion 'width >= 1' failed
cogl_texture_2d_new_with_size: assertion 'width >= 1' failed
cogl_texture_2d_new_with_size: assertion 'width >= 1' failed
cogl_texture_2d_new_with_size: assertion 'width >= 1' failed
cogl_texture_2d_new_with_size: assertion 'width >= 1' failed
cogl_texture_2d_new_with_size: assertion 'width >= 1' failed
cogl_texture_2d_new_with_size: assertion 'width >= 1' failed
cogl_texture_2d_new_with_size: assertion 'width >= 1' failed
cogl_texture_2d_new_with_size: assertion 'width >= 1' failed
display-brightness-ddcutil extension:
Monitor change detected, reloading extension in 5 seconds.
display-brightness-ddcutil extension:
Reload extension
display-brightness-ddcutil extension:
Remove keyboard shortcuts
display-brightness-ddcutil extension:
Adding to panel
display-brightness-ddcutil extension:
Add keyboard shortcuts
Window manager warning: Overwriting existing binding of keysym 65 with keysym 65 (keycode 1a).
display-brightness-ddcutil extension:
ddcutil brief info:
Display 1
   I2C bus:             /dev/i2c-6
   Monitor:             GSM:LG ULTRAGEAR:109NTFAG2459


display-brightness-ddcutil extension:
ddcutil brief info found bus line:
    I2C bus:             /dev/i2c-6
display-brightness-ddcutil extension:
ddcutil reading display state for bus: 6
display-brightness-ddcutil extension:
ddcutil reading display status for bus: 6 is: VCP D6 SNC x01

display-brightness-ddcutil extension:
Reloading widgets
`

**Comment by tuxor1337** [comment]
So, did you try to reproduce this with all other extensions deactivated?

**Comment by Razer0123** [comment]
> So, did you try to reproduce this with all other extensions deactivated?

Yes still happens

`meta_window_unminimize: assertion '!window->override_redirect' failed
meta_window_activate: assertion '!window->override_redirect' failed
meta_window_raise: assertion '!window->override_redirect' failed
meta_window_unminimize: assertion '!window->override_redirect' failed
meta_window_activate: assertion '!window->override_redirect' failed
GNOME Shell started at Mon May 09 2022 22:00:44 GMT+0200 (Ora legale dell’Europa centrale)
Registering session with GDM
meta_window_raise: assertion '!window->override_redirect' failed
meta_window_unminimize: assertion '!window->override_redirect' failed
meta_window_activate: assertion '!window->override_redirect' failed
Timelines with detached actors are not supported
Timelines with detached actors are not supported
meta_monitor_manager_get_logical_monitor_from_number: assertion '(unsigned int) number < g_list_length (manager->logical_monitors)' failed
meta_workspace_get_work_area_for_monitor: assertion 'logical_monitor != NULL' failed
meta_monitor_manager_get_logical_monitor_from_number: assertion '(unsigned int) number < g_list_length (manager->logical_monitors)' failed
meta_workspace_get_work_area_for_monitor: assertion 'logical_monitor != NULL' failed
Timelines with detached actors are not supported
Timelines with detached actors are not supported
Timelines with detached actors are not supported
JS ERROR: Extension hidetopbar@mathieu.bidon.ca: TypeError: Main.layoutManager.primaryMonitor is null
_updateSearchEntryMargin@/home/alessandro/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/panelVisibilityManager.js:323:23
_updateSettingsShowInOverview@/home/alessandro/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/panelVisibilityManager.js:318:14
HideTopBar_PanelVisibilityManager@/home/alessandro/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/panelVisibilityManager.js:71:14
enable@/home/alessandro/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/extension.js:38:18
_callExtensionEnable@resource:///org/gnome/shell/ui/extensionSystem.js:167:32
_enableAllExtensions/<@resource:///org/gnome/shell/ui/extensionSystem.js:607:22
_enableAllExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:606:37
_sessionUpdated@resource:///org/gnome/shell/ui/extensionSystem.js:634:18
_emit@resource:///org/gnome/gjs/modules/core/_signals.js:114:47
_sync@resource:///org/gnome/shell/ui/sessionMode.js:198:14
popMode@resource:///org/gnome/shell/ui/sessionMode.js:166:14
_continueDeactivate@resource:///org/gnome/shell/ui/screenShield.js:532:30
deactivate/<@resource:///org/gnome/shell/ui/screenShield.js:523:44
finish@resource:///org/gnome/shell/gdm/authPrompt.js:604:13
finish@resource:///org/gnome/shell/ui/unlockDialog.js:886:26
deactivate@resource:///org/gnome/shell/ui/screenShield.js:523:26
injections_add/<@/usr/share/gnome-shell/extensions/pop-shell@system76.com/extension.js:485:30
ScreenShield/</<@resource:///org/gnome/shell/ui/screenShield.js:109:57
_emit@resource:///org/gnome/gjs/modules/core/_signals.js:114:47
_convertToNativeSignal@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:152:19

meta_monitor_manager_get_logical_monitor_from_number: assertion '(unsigned int) number < g_list_length (manager->logical_monitors)' failed
meta_workspace_get_work_area_for_monitor: assertion 'logical_monitor != NULL' failed
`

**Comment by IvanTurgenev** [comment]
I have this exact same issue, this are my extension on nixos 22.05
![image](https://user-images.githubusercontent.com/3318444/193366480-61908cc2-5492-4b45-abed-aa20b74209c8.png)

**Comment by IvanTurgenev** [comment]
have to logout for it to fix itself

**Comment by trie-me** [comment]
Gnome 43.2 + Fedora 37.

- Happens when returning from a lock state intermittently. 
  - Suggests a race condition between the firing of the event and the availability of the resource. 
- Recommend adding a try catch at minimum and warning it's not available.
  - this won't fix the issue, but will prevent it from entering the faulted state

Function causing the error:

```javascript
    _updateSearchEntryPadding() {
        if (!_searchEntryBin) return;
        const scale = Main.layoutManager.primaryMonitor.geometry_scale;
        const offset = PanelBox.height / scale; 
        _searchEntryBin.set_style(this._showInOverview ? `padding-top: ${offset}px;` : null);
    }
```

In file `panelVisibilityManager.js`

Journalctl dump

```bash
Dec 11 10:38:06 REDACTED gnome-shell[49022]: JS ERROR: Extension hidetopbar@mathieu.bidon.ca: TypeError: Main.layoutManager.primaryMonitor is null
                                          _updateSearchEntryPadding@/home/REDACTED/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/panelVisibilityManager.js:363:23
                                          _updateSettingsShowInOverview@/home/REDACTED/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/panelVisibilityManager.js:358:14
                                          HideTopBar_PanelVisibilityManager@/home/REDACTED/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/panelVisibilityManager.js:75:14
                                          enable@/home/REDACTED/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/extension.js:38:18
Dec 13 08:26:14 tigre gnome-shell[106052]: JS ERROR: Extension hidetopbar@mathieu.bidon.ca: TypeError: Main.layoutManager.primaryMonitor is null
                                           _updateSearchEntryPadding@/home/REDACTED/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/panelVisibilityManager.js:363:23
                                           _updateSettingsShowInOverview@/home/REDACTED/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/panelVisibilityManager.js:358:14
                                           HideTopBar_PanelVisibilityManager@/home/REDACTED/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/panelVisibilityManager.js:75:14
                                           enable@/home/REDACTED/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/extension.js:38:18
                                           _updateStaticBox@/home/REDACTED/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/panelVisibilityManager.js:320:9
                                           onComplete@/home/REDACTED/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/panelVisibilityManager.js:149:26
                                           _updateStaticBox@/home/REDACTED/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/panelVisibilityManager.js:320:9
                                           onComplete@/home/REDACTED/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/panelVisibilityManager.js:149:26
                                           _updateStaticBox@/home/REDACTED/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/panelVisibilityManager.js:320:9
                                           onComplete@/home/REDACTED/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/panelVisibilityManager.js:149:26
```
